extensions/s3/src/main/java/io/deephaven/extensions/s3/S3AsyncClientFactory.java

```diff
@@ -41,6 +39,25 @@ class S3AsyncClientFactory {
     private static volatile ScheduledExecutorService scheduledExecutor;
 
     static S3AsyncClient getAsyncClient(@NotNull final S3Instructions instructions) {
+        S3AsyncClient s3AsyncClient;
+        try {
+            s3AsyncClient = getAsyncClientBuilder(instructions).build();
```

Comment:
I couldn't do 
```
        try {
            S3AsyncClient client = S3AsyncClient.create();
            client.close();
        } catch (final RuntimeException e) {
            if (instructions.regionName().isEmpty() && e.getMessage().contains("region")) {
                useInstructions = instructions.withRegion("us-east-1");
            }
        }
```
because for the above, when we do `S3AsyncClient.create()`, it internally uses the default `FUTURE_COMPLETION_EXECUTOR` and `scheduledExecutorService` by creating new thread pools, instead of the shared thread pools which we use.